Sources/StreamChat/StateLayer/ChannelListState+Observer.swift

```diff
@@ -117,14 +117,20 @@ extension ChannelListState {
         }
         
         private func linkChannelIfNeeded(_ channel: ChatChannel) async throws {
-            guard !channelListObserver.items.contains(where: { $0.cid == channel.cid }) else { return }
```

Comment:
The reasoning is that looping `LazyCachedMapCollection` triggers loading each DTO separately which can get expensive and loses the point of using cached collection.
Related to PR comment [here](https://github.com/GetStream/stream-chat-swift/pull/3069#discussion_r1521072803)